core/App/components/modals/CameraDisclosureModal.tsx

```diff
@@ -0,0 +1,113 @@
+import { useNavigation } from '@react-navigation/core'
+import { StackNavigationProp } from '@react-navigation/stack'
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Modal, ScrollView, StyleSheet, Text, View, Linking } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+
+import { useTheme } from '../../contexts/theme'
+import { Screens, HomeStackParams } from '../../types/navigators'
+import { testIdWithKey } from '../../utils/testable'
+import Button, { ButtonType } from '../buttons/Button'
+
+import DismissiblePopupModal from './DismissiblePopupModal'
+
+interface CameraDisclosureModalProps {
+  requestCameraUse: () => Promise<boolean>
+}
+
+const CameraDisclosureModal: React.FC<CameraDisclosureModalProps> = ({ requestCameraUse }) => {
+  const { t } = useTranslation()
+  const navigation = useNavigation<StackNavigationProp<HomeStackParams>>()
+  const [modalVisible, setModalVisible] = useState(true)
+  const [showSettingsPopup, setShowSettingsPopup] = useState(false)
+  const [requestInProgress, setRequestInProgress] = useState(false)
+  const { ColorPallet, TextTheme } = useTheme()
+
+  const styles = StyleSheet.create({
+    container: {
+      height: '100%',
+      backgroundColor: ColorPallet.brand.primaryBackground,
+      padding: 20,
+    },
+    messageText: {
+      marginTop: 30,
+    },
+    controlsContainer: {
+      backgroundColor: ColorPallet.brand.primaryBackground,
+      marginTop: 'auto',
+      margin: 20,
+    },
+    buttonContainer: {
+      paddingTop: 10,
+    },
+  })
+
+  const onAllowTouched = async () => {
+    setRequestInProgress(true)
+    const granted = await requestCameraUse()
+    if (!granted) setShowSettingsPopup(true)
```

Comment:
Can you use braces here? Its handy short hand but can lead to errors if someone less experienced picks up JS. One might be tempted to add a line below as part of the `if`.